Common subdirectories: chdkptp-clean/chdk_headers and chdkptp/chdk_headers
diff -u chdkptp-clean/chdkptp.c chdkptp/chdkptp.c
--- chdkptp-clean/chdkptp.c	2021-09-09 19:16:04.964211095 -0400
+++ chdkptp/chdkptp.c	2021-09-09 19:24:02.668705181 -0400
@@ -2658,7 +2658,13 @@
 };
 
 
-static int chdkptp_registerlibs(lua_State *L) {
+static int luaopen_chdkptp(lua_State *L) {
+
+	usb_init();
+	luaopen_lfs(L);
+	luaopen_lbuf(L);
+	luaopen_rawimg(L);	
+
 	/* set up meta table for error object */
 	luaL_newmetatable(L,CHDK_API_ERROR_META);
 	lua_pushcfunction(L,api_error_tostring);
@@ -2728,13 +2734,9 @@
 	/* register signal handlers */
 //	signal(SIGINT, ptpcam_siginthandler);
 	init_timing();
-	usb_init();
 	lua_State *L = luaL_newstate();
 	luaL_openlibs(L);
-	luaopen_lfs(L);
-	luaopen_lbuf(L);
-	luaopen_rawimg(L);
-	chdkptp_registerlibs(L);
+	luaopen_chdkptp(L);
 	int r=exec_lua_string(L,"require('main')");
 	uninit_gui_libs(L);
 	lua_close(L);
diff -u chdkptp-clean/include.mk chdkptp/include.mk
--- chdkptp-clean/include.mk	2021-09-09 19:16:04.954211210 -0400
+++ chdkptp/include.mk	2021-09-09 19:19:56.341542888 -0400
@@ -32,7 +32,7 @@
 EXE_EXTRA=
 
 CC=gcc
-CFLAGS=-DCHDKPTP_OSTYPE=\"$(OSTYPE)\" -Wall
+CFLAGS=-DCHDKPTP_OSTYPE=\"$(OSTYPE)\" -Wall -fPIC
 LDFLAGS=
 #LD=ld
 
Common subdirectories: chdkptp-clean/lfs and chdkptp/lfs
Common subdirectories: chdkptp-clean/lua and chdkptp/lua
Common subdirectories: chdkptp-clean/lua-signal and chdkptp/lua-signal
diff -u chdkptp-clean/Makefile chdkptp/Makefile
--- chdkptp-clean/Makefile	2021-09-09 19:16:04.974210980 -0400
+++ chdkptp/Makefile	2021-09-09 19:18:49.122317721 -0400
@@ -139,14 +139,18 @@
 endif
 
 CHDKPTP_EXE=chdkptp$(EXE_EXTRA)$(EXE)
+CHDKPTP_LIB=chdkptp$(EXE_EXTRA)$(LIB)
 
 EXES=$(CHDKPTP_EXE)
 
-all: $(EXES)
+all: $(EXES) $(LIBS)
 
 SRCS=properties.c ptp.c chdkptp.c lbuf.c liveimg.c rawimg.c luautil.c $(PTPIP_SRCS)
 OBJS=$(SRCS:.c=.o)
 
+$(CHDKPTP_LIB): $(OBJS)
+	$(CC) -shared -o $@ lfs/lfs.o $^ $(LDFLAGS)
+
 $(CHDKPTP_EXE): $(OBJS)
 	$(CC) -o $@ lfs/lfs.o $^ $(LDFLAGS)
 
Common subdirectories: chdkptp-clean/misc and chdkptp/misc
